ACPs/77-reinventing-subnets/README.md

```diff
@@ -109,21 +76,81 @@ type SubnetValidator struct {
     ChangeOwner fx.Owner `json:"changeOwner"`
 }
 
-type RecoverSubnetTx struct {
+type ConvertSubnetTx struct {
     // Metadata, inputs and outputs
     BaseTx
-    // Validator set for the Subnet
+    // ID of the Subnet to transform
+    // Restrictions:
+    // - Must not be the Primary Network ID
+    Subnet ids.ID `json:"subnetID"`
+    // BlockchainID where the Subnet manager lives
+    ChainID ids.ID `json:"chainID"`
+    // Address of the Subnet manager
+    Address []byte `json:"address"`
+    // Initial pay-as-you-go validators for the Subnet
     Validators []SubnetValidator `json:"validators"`
-    // Auth that will be allowing these validators into the network
+    // Authorizes this conversion
     SubnetAuth verify.Verifiable `json:"subnetAuthorization"`
 }
 ```
 
+Once this transaction is accepted, `AddSubnetValidatorTx` is disabled on the Subnet going forward. The only action that the `Owner` key is able to take is removing any Subnet validators that were added using `AddSubnetValidatorTx` previously via `RemoveSubnetValidatorTx`. Unless removed by the `Owner` key, any Subnet Validators added previously with an `AddSubnetValidatorTx` will continue to validate the Subnet until their [`End`](https://github.com/ava-labs/avalanchego/blob/a1721541754f8ee23502b456af86fea8c766352a/vms/platformvm/txs/validator.go#L27) time is reached. Once all Subnet Validators added with `AddSubnetValidatorTx` are no longer in the validator set, the `Owner` key is powerless. `RegisterSubnetValidatorTx` and `SetSubnetValidatorWeightTx` must be used to manage the Subnet's validator set going forward.
+
+The `validationID` for validators added through `ConvertSubnetTx` is defined as the SHA256 hash of the 40 bytes resulting from concatenating the 32 byte `convertSubnetTxID` with the 8 byte `validatorIndex` (index in the `Validators` array within the transaction).
```

Comment:
The `validatorIndex` here being 8 byte value implies that it's a uint64 but the spec for `SubnetConversionData` below specifies the length of validatorData being of length 4 implying uint32. 

Is the intention that this is a uint32 packed as a uint64 to make the full length of the SHA256 pre-image 40?

Reply:
Great catch. All lengths should be 4 bytes (uint32), so the `validatorIndex` should be a uint32 as well, and the pre-image should be 36 bytes. 

Updated now

Reply:
If we're using [ids.Prefix](https://github.com/ava-labs/avalanchego/blob/master/ids/id.go#L97), the uint32 will get cast to a uint64. See UTXOID: https://github.com/ava-labs/avalanchego/blob/master/vms/components/avax/utxo_id.go#L45-L51

Reply:
Hm, in that case it would be 40 bytes with the 8 byte validator index first then?

Reply:
Yeah the ordering of the fields in the spec would need to change if using `ids.Prefix`

Reply:
Thoughts on keeping the (convertSubnetTxID,index) ordering, and not adding the 4 zero bytes to the pre-image?

Changing the serialization for the sake of using an existing helper feels a bit odd. Could always add a new helper if needed

Reply:
Unless anyone is strongly opposed, I suggest we keep this serialization specification as it is currently since it avoid the extra zero bytes and IMO has a more intuitive ordering.

Reply:
I don't really have an opinion here tbh. `ids.Prefix` was floated in earlier convos so I figured we'd just go with that.

If we end up using `ids.Prefix` we should update the spec to match that. If we don't, what you have now is perfectly valid.